### DISCUSSION TITLE
Render Object in Front of  Specific Object

### DISCUSSION THREAD

**Discussion by Psi25** [bnpr/Malt#311] [Q&A] [answered]
Is it possible to render an object in front of a specific object and even give it translucency?

**Comment by pragma37** [comment, marked as answer]
Please, update your question with more details/examples of what you want.

It's not clear what you are asking.

**Reply by Psi25** [reply]
My goal is to make object A which is behind object B to render in front of it.

![2022-06-03 16 55 19](https://user-images.githubusercontent.com/103315144/171823649-4bb56133-c22d-4049-a07f-05a789b55e5e.png)

I was able to achieve this in Eevee using the compositor. Object A rendered in front of object B successfully and I am able to give it translucency/opacity. The problem with this is it uses AOVs of the objects and no realtime preview. I don't know if this is possible in Malt.

![aaa01](https://user-images.githubusercontent.com/103315144/171824063-caac75ae-a2d2-440e-8a5b-4b4fbad637e1.png)

**Reply by pragma37** [reply]
I think the easiest way to achieve that is with the *Depth Offset* Output:

![imagen](https://user-images.githubusercontent.com/36610999/171907598-14320328-cbe0-4b3d-85a1-18148e53eb86.png)

![imagen](https://user-images.githubusercontent.com/36610999/171907768-afd0eda4-1908-451f-b56e-263dc1542cb8.png)

[test_depth_offset.zip](https://github.com/bnpr/Malt/files/8834190/test_depth_offset.zip)

Make sure you download the latest Malt version because there was a bug I had to fix.

**Reply by Psi25** [reply]
It works although it has a problem when a new object is introduced. The only way to fix the problem to re-adjust the offset.
